### PR TITLE
Fix repo_dir cannot export/write mlir file bug

### PR DESCRIPTION
**Pull request by AmosLewis**

### CODE REVIEW

File: shark/parser.py [hunk 1]
```diff
@@ -44,8 +44,7 @@ def dir_file(path):
 parser.add_argument(
     "--repro_dir",
     help="Directory to which module files will be saved for reproduction or debugging.",
-    type=dir_path,
-    default="./repro_dir/",
+    default="./tmp/",
```
Comment (raikonenfnu) [inline shark/parser.py:47]: Can we change this to `./shark_tmp`

Comment (raikonenfnu) [inline shark/parser.py:47]: Can we then add `./shark_tmp` into `.gitignore`?

Comment (AmosLewis) [reply shark/parser.py]: done

Comment (AmosLewis) [reply shark/parser.py]: How

**Comment by monorimet** [comment]
The IREE TempFileSaver doesn't call the mlir/vmfb export functions -- can we put the directory creation in shark/parser.py when it checks the repro_dir path? Otherwise the shark_tmp directory won't be created when only using the --save_temps flag.

**Comment by AmosLewis** [comment]
> The IREE TempFileSaver doesn't call the mlir/vmfb export functions -- can we put the directory creation in shark/parser.py when it checks the repro_dir path? Otherwise the shark_tmp directory won't be created when only using the --save_temps flag.

Done

**Review by raikonenfnu — APPROVED** [review]

**Pull request merged**